onecgiar-pr-server/src/api/results/results-toc-results/results-toc-results.service.ts

```diff
@@ -751,4 +791,132 @@ export class ResultsTocResultsService {
       return this._handlersError.returnErrorRes({ error });
     }
   }
+
+  async saveMapToToc(
+    createResultsTocResultDto: ContributorResultTocResult[],
+    user: TokenDto,
+    result_id: number,
+  ) {
+    try {
+      // * Logic to map multiple WPs to multiple Initiatives Contributors
+      if (createResultsTocResultDto) {
+        for (const contributor of createResultsTocResultDto) {
+          if (!contributor.result_toc_results?.length) {
+            contributor.result_toc_results = [];
+          }
+          for (const rtrc of contributor.result_toc_results) {
+            if (!rtrc?.result_toc_result_id && !rtrc?.toc_result_id) {
+              continue;
+            }
+            await this._resultsTocResultRepository.insert({
+              initiative_ids: contributor?.initiative_id,
+              toc_result_id: rtrc?.toc_result_id,
+              created_by: user.id,
+              last_updated_by: user.id,
+              result_id: result_id,
+              planned_result: contributor?.planned_result,
+              action_area_outcome_id: rtrc?.action_area_outcome_id || null,
+              is_active: true,
+              toc_progressive_narrative:
+                rtrc?.toc_progressive_narrative || null,
+            });
+          }
+        }
+      }
+    } catch (error) {
+      return this._handlersError.returnErrorRes({ error });
+    }
+  }
+
+  private async sendEmailNotification(
+    contributingInit: number[],
+    result_id: number,
+    initSubmitter: number,
+    user: TokenDto,
+  ) {
+    for (const init of contributingInit) {
+      const [initOwner, result, initContributing, initMembers] =
+        await Promise.all([
+          this._clarisaInitiatives.findOne({
+            where: { id: initSubmitter },
+          }),
+          this._resultRepository.findOne({ where: { id: result_id } }),
+          this._clarisaInitiatives.findOne({
+            where: { id: init },
+          }),
+          this._roleByUserRepository.find({
+            where: {
+              initiative_id: init,
+              role: In([3, 4, 5]),
+              active: true,
+            },
+            relations: ['obj_user'],
```

Comment:
@JuanGuzman-io please note that as per [TypeORM documentation](https://github.com/typeorm/typeorm/blob/e7649d2746f907ff36b1efb600402dedd5f5a499/src/find-options/FindOptionsRelations.ts#L43), this way of declaring the relations (using a `string` array), is going to be deprecated on the next version of the dependency, so please use (for this case) `relations: {obj_user: true}` instead